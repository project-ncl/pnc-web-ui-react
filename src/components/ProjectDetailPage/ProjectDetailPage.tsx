import { useEffect } from 'react';
import { useParams } from 'react-router-dom';
import { Card, CardBody } from '@patternfly/react-core';
import { PageLayout } from './../PageLayout/PageLayout';
import { IService, useDataContainer } from '../../containers/DataContainer/useDataContainer';
import { projectService } from '../../services/projectService';
import { DataContainer } from '../../containers/DataContainer/DataContainer';
import { SectionHeader } from '../SectionHeader/SectionHeader';
import { AttributesItems } from '../AttributesItems/AttributesItems';
import { ActionButton } from '../ActionButton/ActionButton';

export const ProjectDetailPage = () => {
  const { projectId } = useParams();

  const dataContainer = useDataContainer(({ requestConfig }: IService) =>
    projectService.getProject({ id: projectId as string }, requestConfig)
  );

  // TODO: Create a better solution than disabling the next line
  useEffect(() => dataContainer.refresh({ requestConfig: {} }), []); // eslint-disable-line react-hooks/exhaustive-deps

  const attributes = [
    {
      name: 'Project URL',
      value: dataContainer.data?.projectUrl && (
        <a href={dataContainer.data?.projectUrl} target="_blank" rel="noopener noreferrer">
          {dataContainer.data?.projectUrl}
        </a>
      ),
    },
    {
      name: 'Issue Tracker URL',
      value: dataContainer.data?.issueTrackerUrl && (
        <a href={dataContainer.data?.issueTrackerUrl} target="_blank" rel="noopener noreferrer">
          {dataContainer.data?.issueTrackerUrl}
        </a>
      ),
    },
    { name: 'Engineering Team', value: dataContainer.data?.engineeringTeam },
    { name: 'Technical Leader', value: dataContainer.data?.technicalLeader },
  ];

  return (
    <DataContainer {...dataContainer} title="Project Details">
      <PageLayout title={dataContainer.data?.name} description={dataContainer.data?.description}>
        <SectionHeader
          actions={
            <ActionButton link="edit" iconType="edit">
              Edit
            </ActionButton>
          }
        />
        <div className="m-b-25">
          <Card>
            <CardBody>
              <AttributesItems attributes={attributes} />
            </CardBody>
          </Card>
        </div>
        <SectionHeader
          actions={
            <ActionButton action={() => console.log('Not implemented yet!')} iconType="create">
              Create
            </ActionButton>
          }
        >
          Build Configs
        </SectionHeader>
        <div className="m-b-25">
          <Card>
            <CardBody>
              <div style={{ backgroundColor: 'yellow', width: '100%', height: '30vh', textAlign: 'center' }}>
                TODO: Add Build Config table here and remove the style object
              </div>
            </CardBody>
          </Card>
        </div>
      </PageLayout>
    </DataContainer>
  );
};
