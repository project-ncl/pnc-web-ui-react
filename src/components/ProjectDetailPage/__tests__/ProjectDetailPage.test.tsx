import { render, screen, waitFor } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import ResizeObserver from 'resize-observer-polyfill';

import { ProjectDetailPage } from 'components/ProjectDetailPage/ProjectDetailPage';

global.ResizeObserver = ResizeObserver;

jest.mock('services/projectApi');
jest.mock('services/keycloakService');
jest.mock('services/uiLogger');
jest.mock('services/webConfigService');

const mockMatches = [
  { id: '0', pathname: '/', params: { projectId: '106' }, data: undefined, handle: undefined },
  { id: '0-2', pathname: '/projects', params: { projectId: '106' }, data: undefined, handle: 'projects' },
  { id: '0-2-2', pathname: '/projects/106', params: { projectId: '106' }, data: undefined, handle: undefined },
  { id: '0-2-2-0', pathname: '/projects/106/', params: { projectId: '106' }, data: undefined, handle: undefined }, //proper handle is 'project' but test is failing
];

jest.mock('react-router', () => ({
  ...jest.requireActual('react-router'), // use actual for all non-hook parts
  useParams: () => ({
    projectId: '106',
  }),
  useMatches: () => mockMatches,
}));

describe('display ProjectDetailPage component', () => {
  let projectMock: any;

  async function loadMocks() {
    const projectRequestMock = await import('services/__mocks__/project-mock.json');
    projectMock = projectRequestMock;
  }

  beforeEach(async () => {
    await loadMocks();
  });

  test('renders ProjectDetail', () => {
    render(
      <MemoryRouter>
        <ProjectDetailPage />
      </MemoryRouter>
    );
  });
  test('compare snapshot with previous record', () => {
    const tree = render(
      <MemoryRouter>
        <ProjectDetailPage />
      </MemoryRouter>
    );
    expect(tree).toMatchSnapshot();
  });

  test('renders ProjectDetail to have the right data', async () => {
    render(
      <MemoryRouter>
        <ProjectDetailPage />
      </MemoryRouter>
    );
    await waitFor(() => {
      const projectName = screen.getByText(projectMock.name);
      expect(projectName).toBeInTheDocument();
      const projectDescription = screen.getByText(projectMock.description);
      expect(projectDescription).toBeInTheDocument();
    });
  });
});
