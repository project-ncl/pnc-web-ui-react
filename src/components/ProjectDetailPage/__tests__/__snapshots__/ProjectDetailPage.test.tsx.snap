// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`display ProjectDetailPage component compare snapshot with previous record 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="pf-v5-c-page__main-breadcrumb"
      />
      <section
        class="pf-v5-c-page__main-section pf-m-light"
      >
        <div
          class="pf-v5-c-content pull-left m-b-15"
        >
          <h1
            class=""
            data-ouia-component-id="OUIA-Generated-Text-4"
            data-ouia-component-type="PF5/Text"
            data-ouia-safe="true"
            data-pf-content="true"
          />
          <p
            class=""
            data-ouia-component-id="OUIA-Generated-Text-5"
            data-ouia-component-type="PF5/Text"
            data-ouia-safe="true"
            data-pf-content="true"
          />
        </div>
        <div
          class="pull-right display-flex align-items-center gap-10"
        >
          <div
            title="Keycloak service is not available."
          >
            <div
              class="disabled-content"
            >
              <a
                aria-disabled="false"
                class="pf-v5-c-button pf-m-primary pf-m-small"
                data-discover="true"
                data-ouia-component-id="OUIA-Generated-Button-primary-3"
                data-ouia-component-type="PF5/Button"
                data-ouia-safe="true"
                href="/edit"
              >
                Edit Project
              </a>
            </div>
          </div>
        </div>
      </section>
      <div
        class="pf-v5-c-divider"
        role="separator"
      />
      <section
        class="pf-v5-c-page__main-section"
      >
        <div
          class="shadow-global bg-w p-global margin-bottom"
        >
          <div
            class=""
          />
        </div>
        <div
          class="shadow-global bg-w border-b-gray p-t-global p-b-global p-l-global p-r-global"
        >
          <div
            class=""
          >
            <div
              class="pf-v5-l-flex pf-m-gap-lg"
            >
              <div
                class=""
              >
                <div
                  class="pf-v5-c-content"
                >
                  <h2
                    class=""
                    data-ouia-component-id="OUIA-Generated-Text-6"
                    data-ouia-component-type="PF5/Text"
                    data-ouia-safe="true"
                    data-pf-content="true"
                  >
                    Build Configs
                  </h2>
                </div>
              </div>
              <div
                class="pf-m-align-right"
              >
                <div
                  title="Keycloak service is not available."
                >
                  <div
                    class="disabled-content"
                  >
                    <a
                      aria-disabled="false"
                      class="pf-v5-c-button pf-m-primary pf-m-small"
                      data-discover="true"
                      data-ouia-component-id="OUIA-Generated-Button-primary-4"
                      data-ouia-component-type="PF5/Button"
                      data-ouia-safe="true"
                      href="/build-configs/create"
                    >
                      Create Build Config
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="shadow-global bg-w p-t-global p-b-global p-l-global p-r-global"
        >
          <div
            class=""
          >
            <div
              class="pf-v5-l-flex pf-m-gap-lg"
            >
              <div
                class=""
              >
                <div
                  class="pf-v5-c-input-group"
                >
                  <div
                    class="pf-v5-c-input-group__item"
                  >
                    <div
                      class="filter-attribute-select"
                    >
                      <button
                        aria-expanded="false"
                        class="pf-v5-c-menu-toggle"
                        data-ouia-component-id="OUIA-Generated-MenuToggle-2"
                        data-ouia-component-type="PF5/MenuToggle"
                        data-ouia-safe="true"
                        type="button"
                      >
                        <span
                          class="pf-v5-c-menu-toggle__text"
                        >
                          Name
                        </span>
                        <span
                          class="pf-v5-c-menu-toggle__controls"
                        >
                          <span
                            class="pf-v5-c-menu-toggle__toggle-icon"
                          >
                            <svg
                              aria-hidden="true"
                              class="pf-v5-svg"
                              fill="currentColor"
                              height="1em"
                              role="img"
                              viewBox="0 0 320 512"
                              width="1em"
                            >
                              <path
                                d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                              />
                            </svg>
                          </span>
                        </span>
                      </button>
                    </div>
                  </div>
                  <div
                    class="pf-v5-c-input-group__item"
                  >
                    <span
                      class="pf-v5-c-form-control form-input"
                    >
                      <input
                        aria-invalid="false"
                        data-ouia-component-id="OUIA-Generated-TextInputBase-3"
                        data-ouia-component-type="PF5/TextInput"
                        data-ouia-safe="true"
                        id="filter-text"
                        placeholder="string | !string | s?ring | st*g"
                        type="text"
                        value=""
                      />
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="shadow-global bg-w border-t-gray"
        >
          <div
            class=""
          />
        </div>
        <div
          class="shadow-global bg-w"
        >
          <div
            class=""
          >
            <div>
              <div
                class="pf-v5-c-pagination pf-m-bottom"
                data-ouia-component-id="OUIA-Generated-Pagination-bottom-2"
                data-ouia-component-type="PF5/Pagination"
                data-ouia-safe="true"
                id="options-menu-bottom-pagination"
                style="--pf-v5-c-pagination__nav-page-select--c-form-control--width-chars: 2;"
              >
                <button
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  class="pf-v5-c-menu-toggle pf-m-plain pf-m-text"
                  data-ouia-component-id="OUIA-Generated-MenuToggle-plainText-2"
                  data-ouia-component-type="PF5/MenuToggle"
                  data-ouia-safe="true"
                  id="options-menu-bottom-toggle"
                  type="button"
                >
                  <span
                    class="pf-v5-c-menu-toggle__text"
                  >
                    <b>
                      0
                       - 
                      0
                    </b>
                     
                    of
                     
                    <b>
                      0
                    </b>
                     
                  </span>
                  <span
                    class="pf-v5-c-menu-toggle__controls"
                  >
                    <span
                      class="pf-v5-c-menu-toggle__toggle-icon"
                    >
                      <svg
                        aria-hidden="true"
                        class="pf-v5-svg"
                        fill="currentColor"
                        height="1em"
                        role="img"
                        viewBox="0 0 320 512"
                        width="1em"
                      >
                        <path
                          d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                        />
                      </svg>
                    </span>
                  </span>
                </button>
                <nav
                  aria-label="Pagination"
                  class="pf-v5-c-pagination__nav"
                >
                  <div
                    class="pf-v5-c-pagination__nav-control pf-m-first"
                  >
                    <button
                      aria-disabled="true"
                      aria-label="Go to first page"
                      class="pf-v5-c-button pf-m-plain pf-m-disabled"
                      data-action="first"
                      data-ouia-component-id="OUIA-Generated-Button-plain-5"
                      data-ouia-component-type="PF5/Button"
                      data-ouia-safe="true"
                      disabled=""
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="pf-v5-svg"
                        fill="currentColor"
                        height="1em"
                        role="img"
                        viewBox="0 0 448 512"
                        width="1em"
                      >
                        <path
                          d="M223.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L319.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L393.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34zm-192 34l136 136c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9L127.9 256l96.4-96.4c9.4-9.4 9.4-24.6 0-33.9L201.7 103c-9.4-9.4-24.6-9.4-33.9 0l-136 136c-9.5 9.4-9.5 24.6-.1 34z"
                        />
                      </svg>
                    </button>
                  </div>
                  <div
                    class="pf-v5-c-pagination__nav-control"
                  >
                    <button
                      aria-disabled="true"
                      aria-label="Go to previous page"
                      class="pf-v5-c-button pf-m-plain pf-m-disabled"
                      data-action="previous"
                      data-ouia-component-id="OUIA-Generated-Button-plain-6"
                      data-ouia-component-type="PF5/Button"
                      data-ouia-safe="true"
                      disabled=""
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="pf-v5-svg"
                        fill="currentColor"
                        height="1em"
                        role="img"
                        viewBox="0 0 256 512"
                        width="1em"
                      >
                        <path
                          d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
                        />
                      </svg>
                    </button>
                  </div>
                  <div
                    class="pf-v5-c-pagination__nav-page-select"
                  >
                    <span
                      class="pf-v5-c-form-control pf-m-disabled"
                    >
                      <input
                        aria-invalid="false"
                        aria-label="Current page"
                        data-ouia-component-id="OUIA-Generated-TextInputBase-4"
                        data-ouia-component-type="PF5/TextInput"
                        data-ouia-safe="true"
                        disabled=""
                        max="0"
                        min="1"
                        type="number"
                        value="0"
                      />
                    </span>
                    <span
                      aria-hidden="true"
                    >
                      of
                       
                      0
                    </span>
                  </div>
                  <div
                    class="pf-v5-c-pagination__nav-control"
                  >
                    <button
                      aria-disabled="true"
                      aria-label="Go to next page"
                      class="pf-v5-c-button pf-m-plain pf-m-disabled"
                      data-action="next"
                      data-ouia-component-id="OUIA-Generated-Button-plain-7"
                      data-ouia-component-type="PF5/Button"
                      data-ouia-safe="true"
                      disabled=""
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="pf-v5-svg"
                        fill="currentColor"
                        height="1em"
                        role="img"
                        viewBox="0 0 256 512"
                        width="1em"
                      >
                        <path
                          d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                        />
                      </svg>
                    </button>
                  </div>
                  <div
                    class="pf-v5-c-pagination__nav-control pf-m-last"
                  >
                    <button
                      aria-disabled="true"
                      aria-label="Go to last page"
                      class="pf-v5-c-button pf-m-plain pf-m-disabled"
                      data-action="last"
                      data-ouia-component-id="OUIA-Generated-Button-plain-8"
                      data-ouia-component-type="PF5/Button"
                      data-ouia-safe="true"
                      disabled=""
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="pf-v5-svg"
                        fill="currentColor"
                        height="1em"
                        role="img"
                        viewBox="0 0 448 512"
                        width="1em"
                      >
                        <path
                          d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34zm192-34l-136-136c-9.4-9.4-24.6-9.4-33.9 0l-22.6 22.6c-9.4 9.4-9.4 24.6 0 33.9l96.4 96.4-96.4 96.4c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l136-136c9.4-9.2 9.4-24.4 0-33.8z"
                        />
                      </svg>
                    </button>
                  </div>
                </nav>
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="pf-v5-c-page__main-breadcrumb"
    />
    <section
      class="pf-v5-c-page__main-section pf-m-light"
    >
      <div
        class="pf-v5-c-content pull-left m-b-15"
      >
        <h1
          class=""
          data-ouia-component-id="OUIA-Generated-Text-4"
          data-ouia-component-type="PF5/Text"
          data-ouia-safe="true"
          data-pf-content="true"
        />
        <p
          class=""
          data-ouia-component-id="OUIA-Generated-Text-5"
          data-ouia-component-type="PF5/Text"
          data-ouia-safe="true"
          data-pf-content="true"
        />
      </div>
      <div
        class="pull-right display-flex align-items-center gap-10"
      >
        <div
          title="Keycloak service is not available."
        >
          <div
            class="disabled-content"
          >
            <a
              aria-disabled="false"
              class="pf-v5-c-button pf-m-primary pf-m-small"
              data-discover="true"
              data-ouia-component-id="OUIA-Generated-Button-primary-3"
              data-ouia-component-type="PF5/Button"
              data-ouia-safe="true"
              href="/edit"
            >
              Edit Project
            </a>
          </div>
        </div>
      </div>
    </section>
    <div
      class="pf-v5-c-divider"
      role="separator"
    />
    <section
      class="pf-v5-c-page__main-section"
    >
      <div
        class="shadow-global bg-w p-global margin-bottom"
      >
        <div
          class=""
        />
      </div>
      <div
        class="shadow-global bg-w border-b-gray p-t-global p-b-global p-l-global p-r-global"
      >
        <div
          class=""
        >
          <div
            class="pf-v5-l-flex pf-m-gap-lg"
          >
            <div
              class=""
            >
              <div
                class="pf-v5-c-content"
              >
                <h2
                  class=""
                  data-ouia-component-id="OUIA-Generated-Text-6"
                  data-ouia-component-type="PF5/Text"
                  data-ouia-safe="true"
                  data-pf-content="true"
                >
                  Build Configs
                </h2>
              </div>
            </div>
            <div
              class="pf-m-align-right"
            >
              <div
                title="Keycloak service is not available."
              >
                <div
                  class="disabled-content"
                >
                  <a
                    aria-disabled="false"
                    class="pf-v5-c-button pf-m-primary pf-m-small"
                    data-discover="true"
                    data-ouia-component-id="OUIA-Generated-Button-primary-4"
                    data-ouia-component-type="PF5/Button"
                    data-ouia-safe="true"
                    href="/build-configs/create"
                  >
                    Create Build Config
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="shadow-global bg-w p-t-global p-b-global p-l-global p-r-global"
      >
        <div
          class=""
        >
          <div
            class="pf-v5-l-flex pf-m-gap-lg"
          >
            <div
              class=""
            >
              <div
                class="pf-v5-c-input-group"
              >
                <div
                  class="pf-v5-c-input-group__item"
                >
                  <div
                    class="filter-attribute-select"
                  >
                    <button
                      aria-expanded="false"
                      class="pf-v5-c-menu-toggle"
                      data-ouia-component-id="OUIA-Generated-MenuToggle-2"
                      data-ouia-component-type="PF5/MenuToggle"
                      data-ouia-safe="true"
                      type="button"
                    >
                      <span
                        class="pf-v5-c-menu-toggle__text"
                      >
                        Name
                      </span>
                      <span
                        class="pf-v5-c-menu-toggle__controls"
                      >
                        <span
                          class="pf-v5-c-menu-toggle__toggle-icon"
                        >
                          <svg
                            aria-hidden="true"
                            class="pf-v5-svg"
                            fill="currentColor"
                            height="1em"
                            role="img"
                            viewBox="0 0 320 512"
                            width="1em"
                          >
                            <path
                              d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                            />
                          </svg>
                        </span>
                      </span>
                    </button>
                  </div>
                </div>
                <div
                  class="pf-v5-c-input-group__item"
                >
                  <span
                    class="pf-v5-c-form-control form-input"
                  >
                    <input
                      aria-invalid="false"
                      data-ouia-component-id="OUIA-Generated-TextInputBase-3"
                      data-ouia-component-type="PF5/TextInput"
                      data-ouia-safe="true"
                      id="filter-text"
                      placeholder="string | !string | s?ring | st*g"
                      type="text"
                      value=""
                    />
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="shadow-global bg-w border-t-gray"
      >
        <div
          class=""
        />
      </div>
      <div
        class="shadow-global bg-w"
      >
        <div
          class=""
        >
          <div>
            <div
              class="pf-v5-c-pagination pf-m-bottom"
              data-ouia-component-id="OUIA-Generated-Pagination-bottom-2"
              data-ouia-component-type="PF5/Pagination"
              data-ouia-safe="true"
              id="options-menu-bottom-pagination"
              style="--pf-v5-c-pagination__nav-page-select--c-form-control--width-chars: 2;"
            >
              <button
                aria-expanded="false"
                aria-haspopup="listbox"
                class="pf-v5-c-menu-toggle pf-m-plain pf-m-text"
                data-ouia-component-id="OUIA-Generated-MenuToggle-plainText-2"
                data-ouia-component-type="PF5/MenuToggle"
                data-ouia-safe="true"
                id="options-menu-bottom-toggle"
                type="button"
              >
                <span
                  class="pf-v5-c-menu-toggle__text"
                >
                  <b>
                    0
                     - 
                    0
                  </b>
                   
                  of
                   
                  <b>
                    0
                  </b>
                   
                </span>
                <span
                  class="pf-v5-c-menu-toggle__controls"
                >
                  <span
                    class="pf-v5-c-menu-toggle__toggle-icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="pf-v5-svg"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      viewBox="0 0 320 512"
                      width="1em"
                    >
                      <path
                        d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                      />
                    </svg>
                  </span>
                </span>
              </button>
              <nav
                aria-label="Pagination"
                class="pf-v5-c-pagination__nav"
              >
                <div
                  class="pf-v5-c-pagination__nav-control pf-m-first"
                >
                  <button
                    aria-disabled="true"
                    aria-label="Go to first page"
                    class="pf-v5-c-button pf-m-plain pf-m-disabled"
                    data-action="first"
                    data-ouia-component-id="OUIA-Generated-Button-plain-5"
                    data-ouia-component-type="PF5/Button"
                    data-ouia-safe="true"
                    disabled=""
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="pf-v5-svg"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      viewBox="0 0 448 512"
                      width="1em"
                    >
                      <path
                        d="M223.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L319.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L393.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34zm-192 34l136 136c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9L127.9 256l96.4-96.4c9.4-9.4 9.4-24.6 0-33.9L201.7 103c-9.4-9.4-24.6-9.4-33.9 0l-136 136c-9.5 9.4-9.5 24.6-.1 34z"
                      />
                    </svg>
                  </button>
                </div>
                <div
                  class="pf-v5-c-pagination__nav-control"
                >
                  <button
                    aria-disabled="true"
                    aria-label="Go to previous page"
                    class="pf-v5-c-button pf-m-plain pf-m-disabled"
                    data-action="previous"
                    data-ouia-component-id="OUIA-Generated-Button-plain-6"
                    data-ouia-component-type="PF5/Button"
                    data-ouia-safe="true"
                    disabled=""
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="pf-v5-svg"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      viewBox="0 0 256 512"
                      width="1em"
                    >
                      <path
                        d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
                      />
                    </svg>
                  </button>
                </div>
                <div
                  class="pf-v5-c-pagination__nav-page-select"
                >
                  <span
                    class="pf-v5-c-form-control pf-m-disabled"
                  >
                    <input
                      aria-invalid="false"
                      aria-label="Current page"
                      data-ouia-component-id="OUIA-Generated-TextInputBase-4"
                      data-ouia-component-type="PF5/TextInput"
                      data-ouia-safe="true"
                      disabled=""
                      max="0"
                      min="1"
                      type="number"
                      value="0"
                    />
                  </span>
                  <span
                    aria-hidden="true"
                  >
                    of
                     
                    0
                  </span>
                </div>
                <div
                  class="pf-v5-c-pagination__nav-control"
                >
                  <button
                    aria-disabled="true"
                    aria-label="Go to next page"
                    class="pf-v5-c-button pf-m-plain pf-m-disabled"
                    data-action="next"
                    data-ouia-component-id="OUIA-Generated-Button-plain-7"
                    data-ouia-component-type="PF5/Button"
                    data-ouia-safe="true"
                    disabled=""
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="pf-v5-svg"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      viewBox="0 0 256 512"
                      width="1em"
                    >
                      <path
                        d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                      />
                    </svg>
                  </button>
                </div>
                <div
                  class="pf-v5-c-pagination__nav-control pf-m-last"
                >
                  <button
                    aria-disabled="true"
                    aria-label="Go to last page"
                    class="pf-v5-c-button pf-m-plain pf-m-disabled"
                    data-action="last"
                    data-ouia-component-id="OUIA-Generated-Button-plain-8"
                    data-ouia-component-type="PF5/Button"
                    data-ouia-safe="true"
                    disabled=""
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="pf-v5-svg"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      viewBox="0 0 448 512"
                      width="1em"
                    >
                      <path
                        d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34zm192-34l-136-136c-9.4-9.4-24.6-9.4-33.9 0l-22.6 22.6c-9.4 9.4-9.4 24.6 0 33.9l96.4 96.4-96.4 96.4c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l136-136c9.4-9.2 9.4-24.4 0-33.8z"
                      />
                    </svg>
                  </button>
                </div>
              </nav>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
